Makefile

```diff
@@ -12,6 +12,41 @@ VERSION         := $(shell grep 'Version:' $(SPEC_FILE) | awk '{printf "%s", $$2
 # Default target
 all: build test
 
+#
+# OpenAPI Code Generation
+#
+OPENAPI_SPEC := api/openapi/owner-server.yaml
+GENERATED_FILE := api/openapi/generated.go
+
+.PHONY: validate-openapi
+validate-openapi:
+	@echo "Validating OpenAPI specification..."
+	@command -v oapi-codegen >/dev/null 2>&1 || command -v $$(go env GOPATH)/bin/oapi-codegen >/dev/null 2>&1 || { \
+		echo "Error: oapi-codegen not found. Please install it first:"; \
+		echo "  go install github.com/oapi-codegen/oapi-codegen/v2/cmd/oapi-codegen@latest"; \
+		exit 1; \
+	}
+	@echo "OpenAPI spec validation (oapi-codegen validates during generation)"
+
+.PHONY: generate-api
+generate-api: validate-openapi
+	@echo "Generating Go types and server code from OpenAPI specification..."
+	$$(command -v oapi-codegen || echo $$(go env GOPATH)/bin/oapi-codegen) -package openapi -generate types,chi-server -o $(GENERATED_FILE) $(OPENAPI_SPEC)
```

Comment:
It seems that the generated file is dependent on the openapi_spec file.  If that's the case then if the openapi_spec file is modified do we want to re-generate the generated code?

If that's the case I'd add GENERATED_FILE to the list of the dependencies for the build target:
```
build: $(GENERATED_FILE) tidy fmt vet
```
Then make the phony generate-api rule into a real dependency rule for GENERATED_FILE:
```
$(GENERATED_FILE): $(OPENAPI_SPEC) validate-openapi
   blab...
```
You can keep the phony generate-api rule, but it would just be a dependency on $(GENERATED_FILE) with no body
```
.PHONY: generate-api
generate-api: $(GENERATED_FILE)
```

Again this is only useful if you want "make build" to re-generate the code if the spec file has changed.